### PR TITLE
Update FedAuth test

### PR DESCRIPTION
**Pull request by helderjs**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: test/int/federated_auth_test.go [hunk 1]
```diff
@@ -150,58 +150,14 @@ var _ = Describe("AtlasFederatedAuth test", Label("AtlasFederatedAuth", "federat
...
-						DeleteRoleMapping(ctx, originalFederationSettings.GetId(), roleMapping.GetId(), orgID).
-						Execute()
-					Expect(err).ToNot(HaveOccurred())
+			roleMappings := make([]akov2.RoleMapping, 0, len(fedAuth.Spec.RoleMappings))
+			for _, roleMap := range fedAuth.Spec.RoleMappings {
+				if roleMap.ExternalGroupName != "ako_team" {
+					roleMappings = append(roleMappings, roleMap)
 				}
 			}
+			fedAuth.Spec.RoleMappings = roleMappings
```
Comment (helderjs) [inline test/int/federated_auth_test.go:160]: I replaced the clean-up logic to reflect the operator ability to add/edit/remove role mappings

Comment (s-urbaniak) [reply test/int/federated_auth_test.go:160]: can you please add a comment that makes it clear that this is changing existing role mappings?

**Review by igor-karpukhin — APPROVED** [review]

File: test/int/federated_auth_test.go (hunk 1 above)
Comment (helderjs) [reply test/int/federated_auth_test.go:160]: @s-urbaniak this PR just adjusts the broken part of the test to better reflect what we want to achieve with it.
A quick look into the full test case shows:
* Retrieve of current config
* Build a FedAuth CR from the current config and add a new role mapping
* Successfully reconcile it
* Modify the CR to remove added role mapping (for test)
* Successfully reconcile it

Comment (s-urbaniak) [reply test/int/federated_auth_test.go:160]: @helderjs sorry for being imprecise, I meant commenting in the code not here 🙏

Comment (helderjs) [reply test/int/federated_auth_test.go]: @s-urbaniak done

**Pull request merged**